src/rust/node-identifier/tests/integration_test.rs

```diff
@@ -0,0 +1,239 @@
+
+                    let parent_to_child_edge = identified_graph
+                        .edges
+                        .get(parent_process.get_node_key())
+                        .iter()
+                        .flat_map(|edge_list| edge_list.edges.iter())
+                        .find(|edge| edge.to_node_key == child_process.get_node_key())
+                        .expect("missing edge from parent to child");
+
+                    parent_to_child_edge.edge_name == "children"
+                } else {
+                    false
+                }
+            },
+        );
+
+        tracing::info!("consuming kafka messages for 30s");
+        assert!(
+            tokio::time::timeout(Duration::from_millis(30000), contains_expected)
+                .await
+                .expect("failed to consume expected message within 30s")
+        );
+    });
+
+    // wait for the kafka consumer to start consuming
+    tracing::info!("waiting for kafka consumer to report ready");
+    rx.await
+        .expect("failed to receive notification that consumer is consuming");
+
+    let log_event: Bytes = r#"<Event xmlns='http://schemas.microsoft.com/win/2004/08/events/event'><System><Provider Name='Microsoft-Windows-Sysmon' Guid='{5770385F-C22A-43E0-BF4C-06F5698FFBD9}'/><EventID>1</EventID><Version>5</Version><Level>4</Level><Task>1</Task><Opcode>0</Opcode><Keywords>0x8000000000000000</Keywords><TimeCreated SystemTime='2019-07-24T18:05:14.402156600Z'/><EventRecordID>550</EventRecordID><Correlation/><Execution ProcessID='3324' ThreadID='3220'/><Channel>Microsoft-Windows-Sysmon/Operational</Channel><Computer>DESKTOP-FVSHABR</Computer><Security UserID='S-1-5-18'/></System><EventData><Data Name='RuleName'></Data><Data Name='UtcTime'>2019-07-24 18:05:14.399</Data><Data Name='ProcessGuid'>{87E8D3BD-9DDA-5D38-0000-0010A3941D00}</Data><Data Name='ProcessId'>5752</Data><Data Name='Image'>C:\Windows\System32\cmd.exe</Data><Data Name='FileVersion'>10.0.10240.16384 (th1.150709-1700)</Data><Data Name='Description'>Windows Command Processor</Data><Data Name='Product'>Microsoft� Windows� Operating System</Data><Data Name='Company'>Microsoft Corporation</Data><Data Name='OriginalFileName'>Cmd.Exe</Data><Data Name='CommandLine'>"cmd" /C "msiexec /quiet /i cmd.msi"</Data><Data Name='CurrentDirectory'>C:\Users\grapltest\Downloads\</Data><Data Name='User'>DESKTOP-FVSHABR\grapltest</Data><Data Name='LogonGuid'>{87E8D3BD-99C8-5D38-0000-002088140200}</Data><Data Name='LogonId'>0x21488</Data><Data Name='TerminalSessionId'>1</Data><Data Name='IntegrityLevel'>Medium</Data><Data Name='Hashes'>MD5=A6177D080759CF4A03EF837A38F62401,SHA256=79D1FFABDD7841D9043D4DDF1F93721BCD35D823614411FD4EAB5D2C16A86F35</Data><Data Name='ParentProcessGuid'>{87E8D3BD-9DD8-5D38-0000-00109F871D00}</Data><Data Name='ParentProcessId'>6132</Data><Data Name='ParentImage'>C:\Users\grapltest\Downloads\svchost.exe</Data><Data Name='ParentCommandLine'>.\svchost.exe</Data></EventData></Event>"#.into();
```

Comment:
would prefer this be an `include_bytes!` or pretty-formatted (now that ian has done some fixes) but not a blocker

Reply:
I'll try pretty printing